test/mocha/cancel.js

```diff
@@ -24,6 +24,31 @@ describe("Cancellation", function() {
         });
     });
 
+    specify("shoud be consistent for mid-chain breaking", function () {
```

Comment:
I wouldn't call this "mid-chain". Yes, it cancels while the chain is currently executing, but that's basically what we have `cancel` for.
"mid-chain" means to me that you have a chain `b = a.then(…), c = b.then(…)` and `b` is cancelled.